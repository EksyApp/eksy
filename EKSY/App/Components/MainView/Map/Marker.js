import React, {Component} from 'react'
import MapView from 'react-native-maps'
import { MarkerShape } from "../../../Utils/PropTypeShapes";

//Renders the marker tags on Map
//Clickable marker component opens a modal (MarkerModal) to view marker's content 
class Marker extends Component {
	constructor (props) {
		super(props)

	}

	_handlePress() {
		this.props.onPress(this.props.marker)
	}

	_markerColor(status) {
		switch (status) {
			case 0: return '#FFFF00'
			case 1: return '#00FF00'
		}
		return '#FF0000'
	}

	render () {
		return (
				<MapView.Marker
						coordinate={{latitude: this.props.marker.latitude, longitude: this.props.marker.longitude}}
						pinColor={this._markerColor(this.props.marker.status)}
						onPress={() => {this._handlePress()}}
				>
				</MapView.Marker>
		)
	}


}

Marker.propTypes = {
	marker: MarkerShape
}

export default Marker
