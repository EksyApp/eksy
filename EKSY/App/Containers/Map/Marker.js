import React, {Component} from 'react'
import MapView from 'react-native-maps'
import PropTypes from 'prop-types'
import * as ReduxActions from '../../Actions'
import {Actions} from 'react-native-router-flux'
import configureStore from '../../Store'

class Marker extends Component {
  constructor (props) {
    super(props)
	  this._initStore()
  }

	async _initStore() {
		this.store = await configureStore()
	}


	_handlePress(event) {
		this.store.dispatch(ReduxActions.setMarkerSelected(this.props.data));
		Actions.markerView()
	}

  render () {
    return (
      <MapView.Marker
        coordinate={{latitude: this.props.data.latitude, longitude: this.props.data.longitude}}
        pinColor={this.props.data.color}
        onPress={(event) => {this._handlePress(event)}}
				>
      </MapView.Marker>
    )
  }


}

Marker.propTypes = {
  data: PropTypes.shape({
	  latitude: PropTypes.number.isRequired,
	  longitude: PropTypes.number.isRequired,
	  color: PropTypes.string,
	  text: PropTypes.string,
	  title: PropTypes.string,
	  images: PropTypes.arrayOf(PropTypes.shape({
		  uri: PropTypes.string.isRequired
	  }))
  }).isRequired

}

export default Marker
