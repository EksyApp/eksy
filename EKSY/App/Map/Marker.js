import React, {Component} from 'react'
import MapView from 'react-native-maps'
import styles from './Styles/MapStyles'
import Callout from './Callout'

class Marker {
  constructor(latitude, longitude) {
    this._latitude = latitude;
    this._longitude = longitude;
    this._pinColor = null;
    this._text = "";
    this._markerComp = null;
    this.key = null;
  }

  getLatitude() {
    return this._latitude;
  }

  getLongitude() {
    return this._longitude;
  }

  setColor(color) {
    this._pinColor = color;
  }

  getColor() {
    return this._pinColor;
  }

  setText(text) {
    this._text = text;
  }

  getText() {
    return this._text;
  }

  getComponent() {
    return(
      <MapView.Marker
        coordinate={{latitude: this._latitude, longitude: this._longitude}}
        pinColor={this._pinColor}
        ref={(ref) => this.markerComp = ref}
        key={this.key}
      >
        <MapView.Callout tooltip style={styles.callout}>
          <Callout
            text={this._text}
          />
        </MapView.Callout>
      </MapView.Marker>
    )
  }


}

export default Marker
