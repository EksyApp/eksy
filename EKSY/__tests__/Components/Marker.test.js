import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Marker from "../../App/Components/MainView/Marker";

jest.mock('react-native-fetch-blob', () => {
	return {
		DocumentDir: () => {},
		polyfill: () => {}
	}
})

test('Marker renders correctly', () => {
	const tree = renderer.create(
			<Marker data={{latitude: 1, longitude: 1}} />
	).toJSON();
	expect(tree).toMatchSnapshot();
});