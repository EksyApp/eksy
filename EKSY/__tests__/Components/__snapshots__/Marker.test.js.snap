// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker renders correctly 1`] = `
<AIRMapMarker
  coordinate={
    Object {
      "latitude": 1,
      "longitude": 1,
    }
  }
  image={undefined}
  onPress={[Function]}
  pinColor={undefined}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
      },
      undefined,
    ]
  }
/>
`;
